compositor_pipeline/src/pipeline/whip_whep/handlers.rs

```diff
@@ -0,0 +1,260 @@
+            .status(StatusCode::INTERNAL_SERVER_ERROR)
+            .body(Body::from("Cannot initalize peer connection"))
+            .unwrap();
+    };
+
+    if let Ok(mut connections) = state.input_connections.lock() {
+        if let Some(connection) = connections.get_mut(&input_id) {
+            connection.peer_connection = Some(peer_connection.clone());
+        } else {
+            return Response::builder()
+                .status(StatusCode::NO_CONTENT)
+                .body(Body::from("Peer connection initialization error"))
+                .unwrap();
+        }
+    }
+
+    peer_connection.on_track(Box::new(move |track, _, _| {
+        let track_kind = track.kind();
+        let video_sender_clone = video_sender.clone();
+        let audio_sender_clone = audio_sender.clone();
+        let state_clone = state.clone();
+        let input_id_clone = input_id.clone();
+        let depayloader_clone = depayloader.clone();
+
+        tokio::spawn(async move {
+            if track_kind == RTPCodecType::Video {
+                if let Some(sender) = video_sender_clone {
+                    handle_track(
+                        track,
+                        state_clone,
+                        input_id_clone,
+                        depayloader_clone,
+                        sender,
+                    )
+                    .await;
+                }
+            } else if track_kind == RTPCodecType::Audio {
+                if let Some(sender) = audio_sender_clone {
+                    handle_track(
+                        track,
+                        state_clone,
+                        input_id_clone,
+                        depayloader_clone,
+                        sender,
+                    )
+                    .await;
+                }
+            }
+        });
+        Box::pin(async {})
+    }));
+
+    // Handle ICE connection state changes (logging for debugging)
+    peer_connection.on_ice_connection_state_change(Box::new(move |state| {
+        info!("ICE connection state changed: {state:?}");
+        Box::pin(async {})
+    }));
+
+    // Set the remote SDP offer
+    peer_connection
+        .set_remote_description(RTCSessionDescription::offer(offer).unwrap())
+        .await
+        .unwrap();
+
+    // Create and set the local SDP answer
+    let answer = peer_connection.create_answer(None).await.unwrap();
+
+    let mut gather = peer_connection.gathering_complete_promise().await;
+
+    peer_connection.set_local_description(answer).await.unwrap();
+
+    let _ = gather.recv().await;
+
+    let sdp = peer_connection.local_description().await.unwrap();
+    info!("Sending SDP answer: {sdp:?}");
+
+    Response::builder()
+        .status(StatusCode::CREATED)
+        .header("Content-Type", "application/sdp")
+        .header("Access-Control-Expose-Headers", "Location")
+        .header("Location", format!("/session/{}", id))
+        .body(Body::from(sdp.sdp.to_string()))
+        .unwrap()
```

Comment:
Is it safe?

Reply:
no, it is not, it should be randomized as in [this section of WHIP RFC](https://datatracker.ietf.org/doc/html/draft-ietf-wish-whip#section-5-4.2.1)
I would also consider implementing safety mechanisms as in the rest of _Security considerations_ section